.github/workflows/performance-checks.yml

```diff
@@ -0,0 +1,39 @@
+name: performance checks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - performance/**
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDKs
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: 17
+      - name: Run build
+        run: ./gradlew :performance:performance:build -Pandroid.useAndroidX=true
+
+  unit_tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDKs
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: 17
+      - name: Run unit tests
+        run: ./gradlew :performance:performance:test -Pandroid.useAndroidX=true
```

Comment:
Oh actually is this just running unit tests for the perf project? Maybe not then.

Reply:
Correct, runs for the perf project, I intend to add lots more tests and a sanity build check in the future.